### DISCUSSION TITLE
How to choose declaration order of views in schema.rb?

### DISCUSSION THREAD

**Discussion by LucaMoriNop** [scenic-views/scenic#424] [Q&A]
We are trying to use both scenic and fx to add views and functions to our database, and one of our views is using a function.
Problem is, when we try to load our schema, views are grouped together before functions, causing an error.

Is there a way to force that view to appear after the function declaration in schema.rb?

Sorry if I'm asking on the wrong repo.

**Comment by derekprior** [comment]
What are you using to make sure the function definition finds its way into `schema.rb`? 

Scenic monkey patches rails' schema dumper. I'm guessing whatever you are using for functions does that same. You would want your functions to come before your views. You could try changing the require order of these two gems (reverse how you have them in your gemfile today). I don't love this solution, but... it might do the trick.

**Comment by LucaMoriNop** [comment]
Hello! Thanks for the answer.

The [fx gem](https://github.com/teoljungberg/fx) is handling the dump of functions/triggers into `schema.rb`.
It does have an option to dump them at the beginning of the schema, but sadly it won't work either with what I'm trying to do,
as some functions use views, and vice versa.

I would like to avoid using `structure.sql` if possible, as both gems allow cleaner management of views/functions, but I can understand that my use case is a bit out of scope.
